api/models/env_var.go

```diff
@@ -69,19 +69,21 @@ func (evs EnvVars) ToKubernetesEnvVars() []v1.EnvVar {
 	return kubeEnvVars
 }
 
-// MergeEnvVars merges two environment variables and return the merging result.
-// Both `left` and `right` EnvVars value will be not mutated.
-// `right` EnvVars has higher precedence.
-func MergeEnvVars(left, right EnvVars) EnvVars {
-	envIndexMap := make(map[string]int, len(left)+len(right))
-	for index, ev := range left {
-		envIndexMap[ev.Name] = index
-	}
-	for _, add := range right {
-		if index, exist := envIndexMap[add.Name]; exist {
-			left[index].Value = add.Value
-		} else {
-			left = append(left, add)
+// MergeEnvVars merges multiple sets of environment variables and return the merging result.
+// All the EnvVars passed as arguments will be not mutated.
+// EnvVars to the right have higher precedence.
+func MergeEnvVars(left EnvVars, rightEnvVars ...EnvVars) EnvVars {
```

Comment:
I extended this helper method a little so that it now merges multiple lists of env vars instead of just 2 lists (this helps with the unit tests where we're chaining this function multiple times in order to merge multiple lists of env vars). The same rules of priority apply.